src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -360,6 +358,53 @@ private static void AddProjectFileDependenciesForPackageReference(PackageSpec pr
             }
         }
 
+        private void AddProjectCentralTransitiveDependencyGroupsForPackageReference(PackageSpec project, LockFile lockFile, IEnumerable<RestoreTargetGraph> targetGraphs)
+        {
+            // TargetFrameworkInformations should have only distinct FrameworkNames
+            var centralPackageVersionsPerFramework = project.TargetFrameworks.ToDictionary(tfmi => tfmi.FrameworkName, tfmi => tfmi.CentralPackageVersions);
```

Comment:
This feels unnecessary to me. 
Dictionaries are an overkill from a performance perspective when the set size is small. 
Having more than 2-3 TFMs is extremely uncommon.

Reply:
Updated.